### ISSUE TITLE
When a note contains a file that does not exist, it shows a Joplin icon next to it

### ISSUE THREAD

**Issue by laurent22** [laurent22/joplin#7794]
- Create a note
- Attach a PDF file - the link is displayed with PDF icon (in the viewer)
    
    <img width="501" alt="image" src="https://user-images.githubusercontent.com/1285584/219720633-078ed6f9-8864-4af8-906d-2feb367b8054.png">

- Change the resource ID to something random, so as to make it invalid. The file is now displayed with a Joplin icon, as if it was an internal link:

    <img width="501" alt="image" src="https://user-images.githubusercontent.com/1285584/219720876-b22ae334-1885-461c-9282-d3976c811b72.png">

Click on the link does nothing (prints an error in the console)
 
The link should instead display a cross icon, or something that shows there's an error. Clicking on the link should display an error message "There is no attachment with this ID"

**Comment by Ayyanaruto** [comment]
Hi, I would like like to look into this issue,Please assign it to me

**Comment by xerkhalidox** [comment]
Can you elaborate more on how the error message should be displayed? is it implemented somewhere else before so we will do the same or a popup will be shown or whatever the approach

**Comment by kanhaiya04** [comment]
Hey, I just tried to reproduce the problem

**Comment by kanhaiya04** [comment]
Hi, I just tried to reproduce the problem
![image](https://user-images.githubusercontent.com/93936630/219800765-1e5bb926-418f-429d-a034-e5e45dde8380.png)

After changing the resource ID
![image](https://user-images.githubusercontent.com/93936630/219800870-b92019fd-b1fc-4754-80f4-9b17d0dbe8b1.png)

**The file is displayed with no icon**
Click on the link does nothing

**Comment by deepampriyadarshi** [comment]
Hi @laurent22 I was able to successfully reproduce this issue. It seems like we need error-handlings for some of the hidden cases, in addition to what you have asked for. 

- Considering your case, the Joplin icon is acting as default icon (**which shouldn't happen**) when the resource list is empty (incorrect resourceId or resource missing from the file system) `&&` the given resourceId is in correct format (32 alphanumeric chars).
- Second case is when the resourceId != 32 alphanumeric chars. Here linkreplacement and openitem work in a very weird manner such that nothing breaks, but they create a totally ambiguous link. In openItem parsing of resource url for such links which starts with `:/` and `wrong format` of resource id returns `null`, passing control of code execution to an `else` branch where _Electron's_ `shell.openExternal()` function is called, which is of `void` type. Being of void type, the functional call remains silent in case of wrong resource url format. The reason of no icon displayed here is that such type of urls are being treated as plain external links.
**TLDR:** URLs starting with `:/` and having `wrong resource id format`, fall under no man's land which are allowed to get rendered but don't have any icon since they can't be treated as a resourceURL and clicking on them has no effect as for  _Electron's_ `shell.openExternal()` function, `:/` is an unknown protocol format.

The second case can occur when a user accidentally delete/add some characters from/to the resource id (which can be recovered by pressing undo **before switching notes**). So do you want to have same functionalities for case 2 that you have mentioned (display a cross icon, error message "There is no attachment with this ID")??

**Comment by dpoulton-github** [comment]
@kanhaiya04

In your example do not append _extra characters_ to the link reference that just make the reference too long to be valid.

If you *replace* the last three characters `516` *with* `dfe` you will have a correct length reference to a non-existent file **and the "J" icon will display**.

**Comment by kanhaiya04** [comment]
@dpoulton-github 
I tried replacing the characters not by adding them, and now I am getting the J icon.
Thanks a lot.
